synapse/tests/test_lib_agenda.py

```diff
@@ -407,17 +413,13 @@ async def test_agenda_persistence(self):
         ''' Test we can make/change/delete appointments and they are persisted to storage '''
         with self.getTestDir() as fdir:
 
-            core = mock.Mock()
+            core = mock.AsyncMock()
 
-            def raiseOnBadStorm(q):
+            async def raiseOnBadStorm(q):
                 ''' Just enough storm parsing for this test '''
-                # TODO: Async this and use AsyncMock when Python 3.8+ only
```

Comment:
this is still a valid TODO and something we can act on now

Reply:
The function was async'd and the mock changed to AsyncMock as part of this PR.